hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/aliasmap/InMemoryAliasMap.java

```diff
@@ -320,21 +320,18 @@ static File createSnapshot(InMemoryAliasMap aliasMap) throws IOException {
   private static File getCompressedAliasMap(File aliasMapDir)
       throws IOException {
     File outCompressedFile = new File(aliasMapDir.getParent(), TAR_NAME);
-    BufferedOutputStream bOut = null;
-    GzipCompressorOutputStream gzOut = null;
-    TarArchiveOutputStream tOut = null;
-    try {
-      bOut = new BufferedOutputStream(
-          Files.newOutputStream(outCompressedFile.toPath()));
-      gzOut = new GzipCompressorOutputStream(bOut);
-      tOut = new TarArchiveOutputStream(gzOut);
+
+    try (BufferedOutputStream bOut = new BufferedOutputStream(
+            Files.newOutputStream(outCompressedFile.toPath()));
+         GzipCompressorOutputStream gzOut = new GzipCompressorOutputStream(bOut);
+         TarArchiveOutputStream tOut = new TarArchiveOutputStream(gzOut)){
+
       addFileToTarGzRecursively(tOut, aliasMapDir, "", new Configuration());
-    } finally {
       if (tOut != null) {
         tOut.finish();
       }
```

Comment:
Thanks for the update.
- Before: `tOut.finish()` is called if addFileToTarGzRecursively throws an exception.
- Your patch: `tOut.finish()` is not called if addFileToTarGzRecursively throws an exception.

I think we need to have an extra try-catch clause:
```java:
      try {
        addFileToTarGzRecursively(tOut, aliasMapDir, "", new Configuration());
      } finally {
        tOut.finish();
      }
```
tOut cannot be null in the try-with-resources clause so that we can remove the null check.

Reply:
If we define `tOut` in try-with-resources block, then we don't have access to `tOut` in finally block. I checked `close()` method in `TarArchiveOutputStream` and `finish()` is called there so I think we don't need to call `finish` on `tOut`.

Reply:
https://github.com/apache/hadoop/pull/3027/commits/6a74dd93fac1433b4a968e59b98eb0596de6666a
let me know if there is any problem.